### PR TITLE
feat: create windows vhd publishing file in build pipeline

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Updating the windows vhd build pipeline to produce a json file with information needed for publishing to be used in future build/release pipeline steps

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by marosset** [comment]
I outlined all of the steps for automate vhd publishing in #2528 
I have a proof-of-concept of the entire workflow in a personal azure devops instance

File: packer/generate-windows-vhd-publishing-info.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+required_env_vars=(
+    "CLASSIC_SA_CONNECTION_STRING"
+    "STORAGE_ACCT_BLOB_URL"
+    "VHD_NAME"
+)
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
```
Comment (devigned) [inline packer/generate-windows-vhd-publishing-info.sh]: I think you want, no?
```bash
if [ -z "${v}" ]; then
  # do stuff
fi
```

File: packer/generate-windows-vhd-publishing-info.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+# Get verion info from release notes
+RELEASE_NOTES_FILE=release-notes.txt
+
+if [ ! -f "$RELEASE_NOTES_FILE" ]; then
+    echo "Could not fine release notes file!"
+    exit 1
+fi
+
+windows_version=$(cat release-notes.txt | grep "OS Version" | cut -d ":" -f 2 | tr -d [:space:])
```
Comment (devigned) [inline packer/generate-windows-vhd-publishing-info.sh]: To fix the lint issue perhaps:
```bash
windows_version=$(grep "OS Version" < release-notes.txt | cut -d "c" -f 2 | tr -d [:space:])
```

File: packer/generate-windows-vhd-publishing-info.sh (hunk 1 above)
Comment (marosset) [reply packer/generate-windows-vhd-publishing-info.sh]: "${!v}" expands the of ${v} when v is an enumerator.
I want to make sure the environment vars in the list have values.
It took me a lot of googling to find that and i'm pretty sure this is the behavior i wanted.

Comment (marosset) [reply packer/generate-windows-vhd-publishing-info.sh]: https://stackoverflow.com/questions/14049057/bash-expand-variable-in-a-variable

**Pull request merged**